website/static/js/pages/profile-settings-addons-page.js

```diff
@@ -0,0 +1,186 @@
+        submit();
+    }
+    return false;
+});
+
+var addonEnabledSettings = window.contextVars.addonEnabledSettings;
+for (var i=0; i < addonEnabledSettings.length; i++) {
+       var addonName = addonEnabledSettings[i];
+       if (typeof window.contextVars.addonsWithNodes !== 'undefined' && addonName in window.contextVars.addonsWithNodes) {
+           AddonPermissionsTable.init(window.contextVars.addonsWithNodes[addonName].shortName,
+                                      window.contextVars.addonsWithNodes[addonName].fullName);
+   }
+}
+
+/***************
+* OAuth addons *
+****************/
+var ConnectedProject = function(data) {
+    var self = this;
+    self.title = data.title;
+    self.id = data.id;
+    self.urls = data.urls;
+};
+
+var ExternalAccount = function(data) {
+    var self = this;
+    self.name = data.display_name;
+    self.id = data.id;
+
+    self.connectedNodes = ko.observableArray();
+
+    ko.utils.arrayMap(data.nodes, function(item) {
+        self.connectedNodes.push(new ConnectedProject(item));
+    });
+
+    self.deauthorizeNode = function(node) {
+        console.log(node);
+        var url = node.urls.deauthorize;
+        $.ajax({
+            url: url,
+            type: 'DELETE'
+        }).done(function(data) {
+            self.connectedNodes.remove(node);
+        }).fail(function(xhr, status, error) {
+            Raven.captureMessage('Error deauthorizing node: ' + node.id, {
+                url: url, status: status, error: error
+            });
+        });
+    }
+
+};
+
+var OAuthAddonSettingsViewModel = function(name, displayName) {
+    var self = this;
+    self.name = name;
+    self.properName = displayName;
+    self.accounts = ko.observableArray();
+    self.message = ko.observable('');
+    self.messageClass = ko.observable('');
+
+    self.setMessage = function(msg, cls) {
+        self.message(msg);
+        self.messageClass(cls || '');
+    };
+
+    self.connectAccount = function() {
+        window.oauthComplete = function() {
+            self.updateAccounts();
+            self.setMessage('Add-on successfully authorized. To link this add-on to an OSF project, go to the settings page of the project, enable ' + self.properName + ', and choose content to connect.', '.text-success');
+        };
+        window.open('/oauth/connect/' + self.name + '/');
+    };
+
+    self.askDisconnect = function(account) {
+        bootbox.confirm({
+            title: 'Delete account?',
+            message: '<p class="overflow">' +
+                'Are you sure you want to delete account <strong>' +
+                account.name + '</strong>?' +
+                '</p>',
+            callback: function(confirm) {
+                if (confirm) {
+                    self.disconnectAccount(account);
+                }
+            }
+        });
+    };
+
+    self.disconnectAccount = function(account) {
+        var url = '/api/v1/oauth/accounts/' + account.id + '/';
+        $.ajax({
```

Comment:
For functions that make HTTP requests, it's nice to return the jqXHR, so that the promise can be used.